### DISCUSSION TITLE
Why is the threshold for Large Page Data set at 128kb?

### DISCUSSION THREAD

**Discussion by ryohiy** [vercel/next.js#72792] [Help] [answered]
### Summary

Hi there,

I was wondering why the warning for [Large Page Data](https://nextjs.org/docs/messages/large-page-data) is set at a 128kb threshold. Is there a specific reason for choosing this limit?

I tried looking for more information but couldn’t find anything in the documentation. I appreciate any insights you can provide. Thank you!

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I remember exploring this some years ago: https://github.com/vercel/next.js/discussions/38364#discussioncomment-3090327

It looks like I arrived at the conclusion that it was somewhat arbitrary, yet reasonable. I'll take a stab at it again, but generally, 128kb of page data, amounts to a lot of text to transfer, and a large object allocation when processing.

### Edit

- https://github.com/vercel/next.js/pull/29956

It must have been that PR that lead me to the arbitrary qualifier 😄

**Reply by ryohiy** [reply]
@icyJoseph 

Thank you! I hadn't come across the related discussions, so I appreciate the help! 🚀

**Reply by icyJoseph** [reply]
No problem. You know you can also change this in the next.config.js file, because sometimes, you'll just have a ton of data and there's no way around that, so one could increase the limit, or perhaps, in other situations even reduce it!

**Reply by icyJoseph** [reply]
See here, https://github.com/vercel/next.js/blob/canary/packages%2Fnext%2Fsrc%2Fserver%2Fconfig-shared.ts#L1129

So it is, `largePageDataBytes` within experimental, that you can tweak to your projects requirements:


```js
    largePageDataBytes: 128 * 1000, // 128KB by default
```

**Comment by ryohiy** [comment]
@icyJoseph 
Thank you so much for your kind and detailed response!
I was able to get the information I needed, so I will go ahead and close this.
